HW1/P5/P5_connected_components.py

```diff
@@ -0,0 +1,105 @@
+import pyspark
+from pyspark import SparkContext
+import time
+
+sc = SparkContext()
+sc.setLogLevel('WARN')
+links = sc.textFile ('s3://Harvard-CS205/wikipedia/links-simple-sorted.txt', 32)
+page_names = sc.textFile ('s3://Harvard-CS205/wikipedia/titles-sorted.txt', 32)
+
+def cleanRawLinks(s):
+	separate = s.split(":")
+	child = separate[1].split()
+	return (int(separate[0]), map(int, child))
+
+
+def cleanRawLinksSymmetrically(s):
+	separate = s.split(":")
+	child = separate[1].split()
+	result = []
+	result.append((int(separate[0]), map(int, child)))
+	for c in map(int, child):
+		result.append((c, [int(separate[0])]))
+	return tuple(result)
+
+page_names = page_names.zipWithIndex().map(lambda (n, id): (id + 1, n))
+page_names = page_names.sortByKey().cache()
+Kevin_Bacon = page_names.filter(lambda (K, V): V == 'Kevin_Bacon').collect()
+Kevin_Bacon = Kevin_Bacon[0][0]
+
+Harvard_University = page_names.filter(lambda (K, V): V == 'Harvard_University').collect()
+Harvard_University = Harvard_University[0][0]
+
+source = Kevin_Bacon
+sink = Harvard_University
+
+cleanLinks = links.map(lambda x: cleanRawLinksSymmetrically(x))
+cleanLinks = cleanLinks.map(lambda x: x[0])
+cleanLinks = cleanLinks.reduceByKey(lambda x, y: list(set(x + y)))
+
+cleanLinks.cache()
+
+
+#The below code tests this approach on a small section of the graph
+# f = open('ccLog.txt', 'w')
+# small_test = sc.textFile('P5_small.txt')
+# clean_small_test = small_test.map(lambda x: cleanRawLinksSymmetrically(x))
+# clean_small_test = clean_small_test.map(lambda x: x[0])
+# clean_small_test = clean_small_test.reduceByKey(lambda x, y: list(set(x + y)))
+# clean_small_test.cache()
+#[(u'1', u' 1664968'), (u'2', u' 3 747213 1664968 1691047 4095634 5535664')]
+
+def CC(graph, source, sink):
```

Comment:
Thank you! Did this before I realized it was extra credit -- unfortunately it runs quite slowly and while it worked for a small sample graph I had written up, I didn't run it for an adequate amount of time on AWS (or maybe its just way too slow to even work on AWS) so I don't have the relevant output for it :(.